src/planner/dart/ConfigurationToConfiguration_to_ConfigurationToTSR.cpp

```diff
@@ -90,63 +90,69 @@ ConfigurationToConfiguration_to_ConfigurationToTSR::plan(
       std::const_pointer_cast<TSR>(problem.getGoalTSR()),
       createSampleableBounds(mMetaSkeletonStateSpace, std::move(rng)),
       ik,
-      problem.getMaxSamples());
+      problem.getMaxSamplingTries());
   auto generator = ikSampleable.createSampleGenerator();
 
   auto robot = mMetaSkeleton->getBodyNode(0)->getSkeleton();
 
-  std::vector<MetaSkeletonStateSpace::ScopedState> configurations;
-
-  // Use a ranker
-  ConstConfigurationRankerPtr configurationRanker(mConfigurationRanker);
-  if (!configurationRanker)
+  std::size_t batches = 0;
+  std::size_t maxBatches = problem.getMaxBatches();
+  std::size_t batchSize = problem.getBatchSize();
+  while (batches < maxBatches && generator->canSample())
   {
-    auto nominalState = mMetaSkeletonStateSpace->createState();
-    mMetaSkeletonStateSpace->copyState(startState, nominalState);
-    configurationRanker = std::make_shared<const NominalConfigurationRanker>(
-        mMetaSkeletonStateSpace, mMetaSkeleton, nominalState);
-  }
 
-  // Goal state
-  auto goalState = mMetaSkeletonStateSpace->createState();
+    std::vector<MetaSkeletonStateSpace::ScopedState> configurations;
 
-  // Sample valid configurations first.
-  static const std::size_t maxSamples{100};
-  std::size_t samples = 0;
-  while (samples < maxSamples && generator->canSample())
-  {
-    // Sample from TSR
-    bool sampled = generator->sample(goalState);
+    // Use a ranker
+    ConstConfigurationRankerPtr configurationRanker(mConfigurationRanker);
+    if (!configurationRanker)
+    {
+      auto nominalState = mMetaSkeletonStateSpace->createState();
+      mMetaSkeletonStateSpace->copyState(startState, nominalState);
+      configurationRanker = std::make_shared<const NominalConfigurationRanker>(
+          mMetaSkeletonStateSpace, mMetaSkeleton, nominalState);
+    }
 
-    // Increment even if it's not a valid sample since this loop
-    // has to terminate even if none are valid.
-    ++samples;
+    // Goal state
+    auto goalState = mMetaSkeletonStateSpace->createState();
 
-    if (!sampled)
-      continue;
+    // Sample valid configurations first.
+    std::size_t samples = 0;
+    while (samples < batchSize && generator->canSample())
+    {
+      // Sample from TSR
+      bool sampled = generator->sample(goalState);
 
-    configurations.emplace_back(goalState.clone());
-  }
+      // Increment even if it's not a valid sample since this loop
+      // has to terminate even if none are valid.
+      ++samples;
 
-  if (configurations.empty())
-    return nullptr;
+      if (!sampled)
```

Comment:
Not a fan of using `continue` or `break` unless necessary, as it can make following code more difficult.

In this case, wrapping `configurations.emplace_back()` in an `if(sampled)` can do away with the `continue` with minimal change.